ckanext/datastore/logic/action.py

```diff
@@ -405,6 +433,16 @@ def datastore_search(context, data_dict):
     :param include_total: True to return total matching record count
                           (optional, default: true)
     :type include_total: bool
+    :param total_estimation_threshold: By default, the "total" returned is a
+            precise count of the query results, which can be computationally
+            expensive for large results (e.g. >100,000k rows). By setting this
+            value to non-zero, and the estimated total is above this value,
+            then datastore_search will skip the precise count - it just returns
+            the estimated total. The estimated total is generated by EXPLAIN,
+            which is triggered by Express Loader or DataPusher as part of
+            loading the data. NB Total estimation is not compatible when
+            specifying 'filters' or 'distinct' options. (optional, default: 0)
```

Comment:
as I mentioned above, I find "0 means infinite" really confusing in an API. Can we use None(null) or some other value as the default "always calculate exact" value?

Reply:
Done!